backend/src/test/java/com/speakfluid/backend/entities/ButtonStepTests.java

```diff
@@ -0,0 +1,171 @@
+                "my lower back pain.");
+        chat1 = new WozMessage("response", "here are some possible remedies for lower" +
+                "back pain. would you like me to list them for you?");
+        user2 = new WozMessage("request", "no, i think i need to go to the hospital. how " +
+                "do i go to the nearest one?");
+        chat2 =  new WozMessage("response", "here are a list of your nearest hospitals. " +
+                "choose the one that you would like");
+        user3 = new WozMessage("request", "i will go to the first one thank you.");
+        chat3 = new WozMessage("response", "you're welcome. have a nice day.");
+        user4 = new WozMessage("request","");
+        chat4 = new WozMessage("response","");
+
+
+
+        chatbotMsgs1.add(chat1);
+        chatbotMsgs2.add(chat2);
+        chatbotMsgs3.add(chat3);
+        chatbotMsgs4.add(chat4);
+
+        userMsgs2.add(user2);
+        userMsgs3.add(user3);
+        userMsgs4.add(user4);
+
+
+        dialogue1 = new Dialogue<>(chatbotMsgs1, userMsgs1);
+        dialogue2 = new Dialogue<>(chatbotMsgs2, userMsgs2);
+        dialogue3 = new Dialogue<>(chatbotMsgs3, userMsgs3);
+        dialogue4 = new Dialogue<>(chatbotMsgs4, userMsgs4);
+
+    }
+
+    @AfterEach
+    public void reset() {
+        // because each message is added before each test, we must clear them to prevent it from repeating.
+        userMsgs1.clear();
+        chatbotMsgs1.clear();
+        userMsgs2.clear();
+        chatbotMsgs2.clear();
+        userMsgs3.clear();
+        chatbotMsgs3.clear();
+        chatbotMsgs4.clear();
+        userMsgs4.clear();
+    }
+
+    @Test
+    void testGetStepName(){
+        String expected = "Button";
+        String actual = button.getStepName();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void setZeroScoreAccumulator(){
+        double expected = 0.0;
+        button.runAnalysis(dialogue1);
+        button.setZeroScoreAccumulator();
+        double actual = button.getScoreAccumulator();
+        assertEquals(expected, actual);
+
+    }
+
+    @Test
+    void testGetMaxScore(){
+        double expected = 15.0;
+        double actual = button.getMaxScore();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void testGetScoreAccumulator(){
+        double expected = 0.0;
+        double actual = button.getScoreAccumulator();
+        assertEquals(expected, actual);
+
+    }
+
+    /*the scoreAccumulator is allowed to exceed the maxscore in our program, in which case
+    the capping mechanism in ConfidenceScoreCalculator will revert it to a rate of 95%
+     */
+
+    @Test
+    void testButtonRunAnalysisDialogue1(){
+        button.runAnalysis(dialogue1);
+        double expected = 10.0;
+        double actual = button.getScoreAccumulator();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testButtonRunAnalysisDialogue2(){
```

Comment:
it might be helpful to name these methods as what they are testing, like testButtonRunAnalysisHighScore